test/framework/controlplane_helpers.go

```diff
@@ -154,17 +155,22 @@ type WaitForControlPlaneToBeReadyInput struct {
 // WaitForControlPlaneToBeReady will wait for a control plane to be ready.
 func WaitForControlPlaneToBeReady(ctx context.Context, input WaitForControlPlaneToBeReadyInput, intervals ...interface{}) {
 	By("Waiting for the control plane to be ready")
-	Eventually(func() (bool, error) {
-		controlplane := &controlplanev1.KubeadmControlPlane{}
+	controlplane := &controlplanev1.KubeadmControlPlane{}
+	Eventually(func() (controlplanev1.KubeadmControlPlane, error) {
 		key := client.ObjectKey{
 			Namespace: input.ControlPlane.GetNamespace(),
 			Name:      input.ControlPlane.GetName(),
 		}
 		if err := input.Getter.Get(ctx, key, controlplane); err != nil {
-			return false, err
+			log.Logf("Failed to list the KCP: %v", err)
```

Comment:
As already discussed, since the error is logged if it is non-nil at the end we should not be logging this anymore. 
however, please be sure to wrap the error returned with a message to provide some more context about the error.

Reply:
done